### PR TITLE
Upgrade action/{upload,download}-artifact to v4 [3.1]

### PR DESCRIPTION
**Pull request by levitte**

### CODE REVIEW

**Review by t8m — APPROVED** [review]
This needs an approval ASAP @openssl/committers

**Review by beldmit — APPROVED** [review]
LGTM

**Comment by levitte** [comment]
Merged

92ab7c5ec63b5d748a80f8f4627d4ac8c87ca5da Upgrade action/{upload,download}-artifact to v4

**Pull request closed without merging**
